apis/core/v1alpha2/readinessprovider_types.go

```diff
@@ -42,6 +42,22 @@ type ReadinessProviderSpec struct {
 
 	// Conditions is the set of checks that must be evaluated to true to mark the provider as ready
 	Conditions []ReadinessProviderCondition `json:"conditions"`
+
+	// ServiceAccount represents the service account to be used
+	// to make requests to the API server for evaluating conditions.
+	// If not provided, it uses the default service account
+	// of the readiness provider controller.
```

Comment:
Add: `The default service account may not have appropriate RBAC to evaluate readiness.`

Reply:
Updated description